src/coreclr/interpreter/eeinterp.cpp

```diff
@@ -0,0 +1,111 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+#include <stddef.h>
+#include "corjit.h"
+
+#include "interpreter.h"
+#include "eeinterp.h"
+
+#include <string.h>
+#include <stdio.h>
+
+#ifdef _MSC_VER
+#define INTERP_API
+#else
+#define INTERP_API __attribute__ ((visibility ("default")))
+#endif // _MSC_VER
+
+/*****************************************************************************/
+ICorJitHost* g_interpHost        = nullptr;
+bool         g_interpInitialized = false;
+/*****************************************************************************/
+extern "C" INTERP_API void jitStartup(ICorJitHost* jitHost)
+{
+    if (g_interpInitialized)
+    {
+        return;
+    }
+    g_interpHost = jitHost;
+    // TODO Interp intialization
+    g_interpInitialized = true;
+}
+/*****************************************************************************/
+static CILInterp g_CILInterp;
+extern "C" INTERP_API ICorJitCompiler* getJit()
+{
+    if (!g_interpInitialized)
+    {
+        return nullptr;
+    }
+    return &g_CILInterp;
+}
+
+//****************************************************************************
+CorJitResult CILInterp::compileMethod(ICorJitInfo*         compHnd,
+                                   CORINFO_METHOD_INFO* methodInfo,
+                                   unsigned             flags,
+                                   uint8_t**            entryAddress,
+                                   uint32_t*            nativeSizeOfCode)
+{
+
+    const char *methodName = compHnd->getMethodNameFromMetadata(methodInfo->ftn, nullptr, nullptr, nullptr, 0);
+
+    // TODO: replace this by something like the JIT does to support multiple methods being specified and we don't
+    // keep fetching it on each call to compileMethod
+    const char *methodToInterpret = g_interpHost->getStringConfigValue("AltJit");
+    bool doInterpret = (methodName != NULL && strcmp(methodName, methodToInterpret) == 0);
+    g_interpHost->freeStringConfigValue(methodToInterpret);
+
+    if (!doInterpret)
+    {
+        return CORJIT_SKIPPED;
+    }
+
+    InterpCompiler compiler(compHnd, methodInfo);
+    InterpMethod *pMethod = compiler.CompileMethod();
+
+    int32_t IRCodeSize;
+    int32_t *pIRCode = compiler.GetCode(&IRCodeSize);
+ 
+    // FIXME this shouldn't be here
+    compHnd->setMethodAttribs(methodInfo->ftn, CORINFO_FLG_INTERPRETER);
```

Comment:
@jkotas these are special attributes that are used for communicating details back to the runtime. The JIT does the same thing. See e.g. the CORINFO_FLG_SWITCHED_TO_OPTIMIZED usage. These flags are special and are not set on the `MethodDesc`. I see reporting back that the JIT generated interpreted code using this flag as something in line with how the other JIT flags are used.

Reply:
These special attributes are used for communicating details that that runtime cannot possibly know. They are computed from non-trivial analysis of the IL. This detail is not like that.

The runtime should know upfront that it is about to call JIT to generate interpreter code. This is related to our discussion about the dedicated code manager for interpreter code in the other PR.

Also, there is `getExpectedTargetArchitecture` JIT/EE interface API. This API should return a new unique architecture value that identifies interpreter. It requires runtime to know that we are generating interpreter code upfront as well.

Reply:
> So the interpreter itself decides which methods it can compile and the runtime falls back to regular JIT for the ones that it rejects

`CORJIT_SKIPPED` return code from the JIT is meant for this purpose. The current altjit mechanism uses it to communicate that the aljit was not interested in compiling the method and regular JIT should be used as a fallback.

Reply:
Yes, that's the way it works with the interpreter too. The reason I was mentioning this is that the runtime doesn't know if the AltJIT is another JIT or the interpreter. So the new attribute is a way to communicate that back to the caller that needs to know that without introducing something else for the runtime to recognize that the AltJIT produced interpreter IR byte code. We could come up with possibly many other ways to do that (setting a bit on the code address returned by the interpreter, adding a new API to the JIT to get such info etc.) However, the attribute seemed to be the least intrusive way to achieve that.

Reply:
It is ok for now, but I am not convinced that it is the design we want in the long run.